### ISSUE TITLE
g_signal_emit usage verification

### ISSUE THREAD

**Issue by Therzok** [mono/monodevelop#4874]
We should validate all calls to `GLib.Signal.Emit`/`g_signal_emit` are done on the right thread.

In the case of UI context, we should assert we're on the main thread before triggering it.

Primary focus should be a11y (atkcocoa/managed atkcocoa helpers)
